pkg/server/conn_stmt.go

```diff
@@ -192,12 +200,28 @@ func (cc *clientConn) handleStmtExecute(ctx context.Context, data []byte) (err e
 		// new param bound flag
 		if data[pos] == 1 {
 			pos++
-			if len(data) < (pos + (numParams << 1)) {
-				return mysql.ErrMalformPacket
+			// For client that has query attribute ability, parameter's name will also be sent.
+			// However, it is useless for execute statement, so we ignore it here.
+			if clientHasQueryAttr {
```

Comment:
Should we also set `SessionVars.QueryAttributes` in this branch? Or a statement cannot use related functions if it's executed through `COM_EXECUTE`.

Reply:
I do not think set `SessionVars.QueryAttributes` in this branch is a good idea since query attribute only 'works' in`COM_QUERY`. SessionVars.QueryAttributes has no other purpose beyond being used for related functions.